fabric-fluid-v1/src/main/java/net/fabricmc/fabric/api/fluid/v1/container/FluidContainer.java

```diff
@@ -0,0 +1,92 @@
+package net.fabricmc.fabric.api.fluid.v1.container;
+
+import java.util.Iterator;
+
+import net.minecraft.fluid.Fluid;
+import net.minecraft.fluid.Fluids;
+
+import net.fabricmc.fabric.api.fluid.v1.Action;
+import net.fabricmc.fabric.api.fluid.v1.container.volume.FluidVolume;
+import net.fabricmc.fabric.api.fluid.v1.container.volume.SimpleFluidVolume;
+
+/**
+ * An object that stores fluid.
+ * FluidContainers can hold one or more types of fluid at once.
+ *
+ * @see FluidVolume
+ */
+public interface FluidContainer extends Iterable<FluidVolume> {
+	/**
+	 * Drains a wildcard matched fluid.
+	 * @param amount the amount to drain
+	 * @param action the nature of the transaction
+	 * @return the fluid drained
+	 */
+	default FluidVolume drain(long amount, Action action) {
+		return this.drain(Fluids.EMPTY, amount, action);
+	}
+
+	/**
+	 * Drain an amount of fluid from the container.
+	 *
+	 * @param fluid the specific fluid to drain, or EMPTY for wildcard
```

Comment:
isn't empty for wildcard just a bit weird? Maybe just accept a nullable fluid ~~or even a fluid predicate~~

Reply:
fluid predicate is a viable option